config/yaml_config.go

```diff
@@ -75,19 +77,33 @@ func (a *AgentConfig) loadNetworkYamlConfig(path string) error {
 	// Pull additional parameters from the global config file.
 	a.LogLevel = config.Datadog.GetString("log_level")
 	a.StatsdPort = config.Datadog.GetInt("dogstatsd_port")
+	if config.Datadog.IsSet("bind_host") {
+		a.StatsdHost = config.Datadog.GetString("bind_host")
+	}
 
 	return nil
 }
 
 // Process-specific configuration
-func (a *AgentConfig) loadProcessYamlConfig(path string) error {
-	config.Datadog.AddConfigPath(path)
-	if strings.HasSuffix(path, ".yaml") { // If they set a config file directly, let's try to honor that
-		config.Datadog.SetConfigFile(path)
-	}
+func (a *AgentConfig) loadProcessConfig(iniPath, yamlPath string) error {
+	if iniPath != "" { // Loading legacy agent 5 configuration (.conf)
+		iniConfig, err := legacy.GetAgentConfig(iniPath)
+		if err != nil {
+			return err
+		}
+		legacy.FromAgentConfig(iniConfig)
+	} else if yamlPath != "" { // Loading modern agent 6 configuration (.yaml)
+		config.Datadog.AddConfigPath(yamlPath)
+		// If they set a config file directly, let's try to honor that
+		if strings.HasSuffix(yamlPath, ".yaml") {
+			config.Datadog.SetConfigFile(yamlPath)
```

Comment:
`yml` is sometimes used as an extension.  I see similar code was elsewhere in the agent config parsing, so I guess we're intentionally only looking for `yaml`

Reply:
Yeah, our `yaml` config example is generated with the `.yaml` suffix. I don't think we've run into any cases where a customer has provided `.yml` or another flavor